doc/python-type-hinting.md

```diff
@@ -40,18 +40,16 @@ def get_objects_matching(ids: Sequence[str]) -> dict[int, dict]
 
 ## Optional arguments
 
-When `Optional` is used for function arguments, it means that the argument can have the value `None`, not that the argument is optional. 
-
-For arguments that can be a single type or `None`, use `Optional`:
+For arguments that can be a single type or None, use the shorthand `|`. This is recommended instead of using older `Optional` keyword:
 
 ```python
-def test(self, var: Optional[int]) -> None:
+def test(self, var: [None | int]) -> None:
 ```
 
-For arguments that can be one of multiple types or `None`, use `Union` with `None`:
+For arguments that can be one of multiple types or None, use the shorthand `|`. This is recommended instead of using older `Union` keyword:
 
 ```python
-def test(self, var: Union[None, int, str]) -> None:
+def test(self, var: [None | int | str]) -> None:
```

Comment:
For this one, this comes from the option which we disabled called `implict_optional`. As recommended and following zen of Python "explict is better than implict" is why this is enforced. The original PEP discourages this. https://peps.python.org/pep-0484/#union-types

From the PEP:

A past version of this PEP allowed type checkers to assume an optional type when the default value is None, as in this code:
```py
def handle_employee(e: Employee = None): ...
```

This would have been treated as equivalent to:
```py
def handle_employee(e: Optional[Employee] = None) -> None: ...
```
This is no longer the recommended behavior. Type checkers should move towards requiring the optional type to be made explicit.

Reply:
As this is no longer the recommended behaviour, I understand that I do not need to add a section on this to the documentation?

Reply:
Thanks @MebinAbraham , wasn't aware of this change, happy to ignore this comment @Hajara-I  👍